src/widget/widget.cpp

```diff
@@ -1079,20 +1080,27 @@ void Widget::onFriendUsernameChanged(int friendId, const QString& username)
         return;
 
     QString str = username; str.replace('\n', ' ');
-    str.remove('\r'); str.remove(QChar((char)0)); // null terminator...
+    str.remove('\r');
+    str.remove(QChar()); // null terminator...
     f->setName(str);
 }
 
 void Widget::onFriendDisplayChanged(FriendWidget *friendWidget, Status s)
 {
     contactListWidget->moveWidget(friendWidget, s);
-    int filter = getFilterCriteria();
+    FilterCriteria filter = getFilterCriteria();
     switch (s)
     {
-        case Status::Offline:
-            friendWidget->searchName(searchContactText->text(), filterOffline(filter));
-        default:
-            friendWidget->searchName(searchContactText->text(), filterOnline(filter));
+    case Status::Offline:
+        friendWidget->searchName(searchContactText->text(), filterOffline(filter));
+                                 filterOffline(filter));
```

Comment:
And here we have an example of what can happen, if conflicts are not resolved correctly. Damn I was sure I had fixed that during the nightly rebase session… :)

I'll rebase it again and remove the duplicate `filterOffline(filter));`, but you'll have to rebase your branch as well - which is why I like to avoid that. Anyway, it's fine in this case and makes every commit compilable at least.